### ISSUE TITLE
Difference in `check` and `server` config discovery

### ISSUE THREAD

**Issue by elementalvoid** [rvben/rumdl#131]
Firstly, thanks for pulling in my fix for the LSP user/XDG config fallback not working. That is a good start, and gets me moving. However, what I'm finding is that the LSP config loading behaves significantly differently from the way the `check` command's loading works leading to variable results.

The first example was the fallback, but then, I tried it in a Python project today (which has a `pyproject.toml`). The LSP's config loading bailed as soon as it saw that file even though it didn't have any rumdl config in it.

## Expected Results

LSP `server` and `check` gives same results.

## Actual Results

`check` command returns success and LSP `server` warns about line length.

Below I'm running a local build which includes the unreleased cherry pick you did for the XDG fallback...

Running `check` it behaves as I wish; I have line length disabled and no warnings are given:
```bash
❯ ll pyproject.toml
Permissions Size User       Group Date Modified Git Name
.rw-r--r--@  508 matt.klich staff  1 Nov 22:42   -M  pyproject.toml

❯ grep -c rumdl pyproject.toml
0

❯ ~/code/github/rumdl/target/release/rumdl config file
/Users/matt.klich/.config/rumdl/rumdl.toml

❯ ~/code/github/rumdl/target/release/rumdl config get global.disable
global.disable = ["MD013", "MD041"] [from .rumdl.toml]

❯ ~/code/github/rumdl/target/release/rumdl check -verbose README.md
Enabled rules:

Exclude patterns: []
Processing file: README.md
Linting took: 6.993125ms
Total processing time for README.md: 7.773541ms

Success: No issues found in 1 file (8ms)
```

(🤔 Side note: `config` prints only the file name `.rumdl.toml` and not the full path `~/.config/rumdl/rumdl.toml` which can be confusing.)

But via LSP I get warnings about line length and my LSP logs show the following:

```
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp] Starting rumdl Language Server Protocol server
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp::server] Initializing rumdl Language Server
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp::server] Workspace root: /Users/matt.klich/code/project-name
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Current working directory: Ok("/Users/matt.klich/code/project-name")
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] No explicit config_path provided, will search default locations
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Checking for user configuration in: /Users/matt.klich/.config/rumdl
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Found user configuration at: /Users/matt.klich/.config/rumdl/rumdl.toml
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Loading user configuration file:/Users/matt.klich/.config/rumdl/rumdl.toml
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Searching for config in: /Users/matt.klich/code/project-name
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Found config file: /Users/matt.klich/code/project-name/pyproject.toml
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Loading discovered config file: /Users/matt.klich/code/project-name/pyproject.toml
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp::server] Loaded rumdl config from: /Users/matt.klich/.config/rumdl/rumdl.toml
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp::server] rumdl Language Server v0.0.170 initialized (built: 2025-11-01 07:03:37 UTC, binary: /Users/matt.klich/code/github/rumdl/target/release/rumdl)
[2025-11-02T04:42:47Z INFO  rumdl_lib::lsp::server] Working directory: /Users/matt.klich/code/project-name
[2025-11-02T04:42:47Z DEBUG rumdl_lib::lsp::server] Config cache miss for directory: /Users/matt.klich/code/project-name, searching for config...
[2025-11-02T04:42:47Z DEBUG rumdl_lib::lsp::server] Found config file: /Users/matt.klich/code/project-name/pyproject.toml
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Current working directory: Ok("/Users/matt.klich/code/project-name")
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Explicit config_path provided: Some("/Users/matt.klich/code/project-name/pyproject.toml")
[2025-11-02T04:42:47Z DEBUG rumdl_lib::config] [rumdl-config] Trying to load config file: pyproject.toml
[2025-11-02T04:42:47Z DEBUG rumdl_lib::lsp::server] Config cache hit for directory: /Users/matt.klich/code/project-name/pyproject.toml)
```

I have verbose logs enabled and we see that it found my XDG config. It also found the `pyproject.toml`. And because it found that it ignores the XDG config. BUT! The `check` command does not ignore the XDG config so I get different results.

---

This is a more clear example of the issue. But it was exactly this that lead me to look at a possible `server config` subcommand. It wasn't that I really wanted to see the config, it was that I wasn't seeing what I expected.

**Comment by rvben** [comment]
Hi @elementalvoid,

Thanks for raising this issue. While some improvements have been made for 0.0.175, the full fix will be expected for the 0.0.176 release.

**Comment by elementalvoid** [comment]
👍 I just checked locally with a9f0fd9 and can confirm that it (a) ignores an empty `pyproject.toml` and (b) merges with global config if not empty.

Thanks! This helps a ton.

**Comment by elementalvoid** [comment]
I'd like to ask a question that is not LSP specific though... If it's expected, cool. If not, I can open a new issue.

---

Testing with very simple configs, I notice if my global config disables a rule, the local config cannot re-enable it.

`~/.config/rumdl/rumdl.toml`
```toml
[global]
disable = ["MD013", "MD041"]
```

`pyproject.toml`
```toml
[tool.rumdl]
enable = ["MD013"]
line-length = 30
```

The config loads like I expect:
```
❯ rumdl config get global.enable
global.enable = ["MD013"] [from pyproject.toml]

❯ rumdl config get global.disable
global.disable = ["MD013", "MD041"] [from .rumdl.toml]

❯ rumdl config get MD013.line-length
MD013.line-length = 30 [from pyproject.toml]
```

So, the merging looks correct. But when I `check` a file the `enable` doesn't override disable.

```
❯ rumdl check -verbose README.md
Enabled rules:

Exclude patterns: []
Processing file: README.md
Linting took: 8.198625ms
Total processing time for README.md: 8.911833ms

Success: No issues found in 1 file (9ms)
```

Is that the expected / desired behavior?

**Comment by rvben** [comment]
Hi @elementalvoid,

Thanks for pointing me to this additional issue.
This behavior is not as intended, the project level config should override the global config. I'm investigating.
